babel/numbers.py

```diff
@@ -325,11 +325,10 @@ def _get_numbering_system(locale: Locale, numbering_system: Literal["default"] |
 
 
 def _get_number_symbols(
-    locale: Locale | str | None,
+    parsed_locale: Locale,
```

Comment:
In `_format_currency_long_name`, the parameter name is still `locale`. I'd keep it here as well perhaps?
```suggestion
    locale: Locale,
```